src/utils/tx-history-filter.ts

```diff
@@ -0,0 +1,277 @@
+  }
+
+  return Object.keys(filter).length > 0 ? filter : undefined
+}
+
+export const parseMultisigFilterUrlQuery = ({
+  execution_date__gte,
+  execution_date__lte,
+  value,
+  to,
+  nonce,
+  executed,
+}: ParsedUrlQuery): MultisigTxFilter => {
+  const filter = {
+    ...(isString(execution_date__gte) && {
+      execution_date__gte: _getDateISO(execution_date__gte),
+    }),
+    ...(isString(execution_date__lte) && {
+      execution_date__lte: _getDateISO(execution_date__lte),
+    }),
+    // TODO: Relevant to token decimals?
+    ...(isString(value) && { value: safeParseUnits(value, 18)?.toString() }),
+    ...(isString(to) && { to }),
+    ...(isString(nonce) && { nonce }),
+    ...(isString(executed) && { executed }),
+  }
+
+  return Object.keys(filter).length > 0 ? filter : undefined
+}
+
+export const parseModuleFilterUrlQuery = ({ to, module }: ParsedUrlQuery): MultisigTxFilter => {
+  const filter = {
+    ...(isString(to) && { to }),
+    ...(isString(module) && { module }),
+  }
+
+  return Object.keys(filter).length > 0 ? filter : undefined
+}
+
+/**
+ * Parses for form filters. We ensure that string values have at least 1 character
+ * and convert Date objects to ISO strings
+ */
+
+export const isDirty = (value: unknown): value is string => {
+  return isString(value) && value.length > 0
+}
+
+export const parseIncomingFilterFormData = ({
+  execution_date__gte,
+  execution_date__lte,
+  value,
+  token_address,
+}: TxFilterFormState): IncomingTxFilter => {
+  const filter = {
+    ...(isDate(execution_date__gte) && {
+      execution_date__gte: _getDateISO(execution_date__gte),
+    }),
+    ...(isDate(execution_date__lte) && {
+      execution_date__lte: _getDateISO(execution_date__lte),
+    }),
+    // TODO: Relevant to token decimals?
+    ...(isDirty(value) && { value: safeParseUnits(value, 18)?.toString() }),
+    ...(isDirty(token_address) && { token_address }),
+  }
+
+  return Object.keys(filter).length > 0 ? filter : undefined
+}
+
+export const parseMultisigFilterFormData = ({
+  execution_date__gte,
+  execution_date__lte,
+  value,
+  to,
+  nonce,
+}: TxFilterFormState): MultisigTxFilter => {
+  const filter = {
+    ...(isDate(execution_date__gte) && {
+      execution_date__gte: _getDateISO(execution_date__gte),
+    }),
+    ...(isDate(execution_date__lte) && {
+      execution_date__lte: _getDateISO(execution_date__lte),
+    }),
+    // TODO: Relevant to token decimals?
+    ...(isDirty(value) && { value: safeParseUnits(value, 18)?.toString() }),
+    ...(isDirty(to) && { to }),
+    ...(isDirty(nonce) && { nonce }),
+    // There is no `executed` field in the form
+  }
+
+  return Object.keys(filter).length > 0 ? filter : undefined
+}
+
+export const parseModuleFilterFormData = ({ to, module }: TxFilterFormState): MultisigTxFilter => {
+  const filter = {
+    ...(isDirty(to) && { to }),
+    ...(isDirty(module) && { module }),
+  }
+
+  return Object.keys(filter).length > 0 ? filter : undefined
+}
```

Comment:
I'll be honest, this is... not very legible and at the same time very verbose.

Reply:
How else would you recommend parsing the individual objects?